src/main/java/nl/opengeogroep/safetymaps/server/stripes/GeoserverActionBean.java

```diff
@@ -0,0 +1,145 @@
+                ResultSetHandler<String> handleQuery = new ResultSetHandler<String>() {
+                    public String handle(ResultSet rs) throws SQLException {
+                        if (!rs.next()) {
+                            return null;
+                         }
+                         return rs.getString(1);
+                    }
+                };
+                String name = geoserverUrl.split("\\?", 2)[0] + "?";
+                String auth = DB.qr().query("select auth from organisation.wms where url = ?", handleQuery, name);
+
+                connection.setRequestMethod("GET");
+                if (auth != null) {
+                    //String encoding = Base64.getEncoder().encodeToString((auth).getBytes());
+                    connection.setRequestProperty("Authorization", "Basic " + auth);
+                }
+
+                InputStream content = (InputStream)connection.getInputStream();
+                Resolution res = pngStreamingResolution(content);
+
+                return res;
+            }
+
+            return new ErrorResolution(HttpServletResponse.SC_NOT_FOUND, "Not found: " + getContext().getRequest().getRequestURI());
+        } catch(Exception e) {
+            return new StreamingResolution("application/json", logExceptionAndReturnJSONObject(log, "Error on " + getContext().getRequest().getRequestURI(), e).toString(0));
+        }
+    }
+
+    /**
+     * Creates a png streaming resolution from a input stream
+     * @param content
+     * @return
+     */
+    private Resolution pngStreamingResolution(final InputStream content) {
+        return new Resolution() {
+            @Override
+            public void execute(HttpServletRequest request, HttpServletResponse response) throws Exception {
+                response.setContentType("image/png");
+                response.setHeader("content-disposition", "inline; filename=geoserverlayer.png");
+
+                OutputStream out = response.getOutputStream();
+                
+                int c;
+                while ((c = content.read()) != -1) {
+                    out.write(c);
+                }
+
+                out.flush();
+                out.close();
+            }
+        };
+    }
+    
+    /**
+     * Create a paramatized url string from params in a request
+     * @param req the request
+     * @return paramatized url string
+     */
+    private String handleParams(HttpServletRequest req) {
+        String out = "";
+        Enumeration<String> parameterNames = req.getParameterNames();
+ 
+        while (parameterNames.hasMoreElements()) {
+            String paramName = parameterNames.nextElement();
+            String[] paramValues = req.getParameterValues(paramName);
+
+            out += "&" + paramName + "=";
+            for (int i = 0; i < paramValues.length; i++) {
+                String paramValue = paramValues[i];
+                out += paramValue;
```

Comment:
Concatenating multiple parameter values is not how it works. This turns ?a=1&a=2&a=3 into a=123